Code_Exercises/Exercise_06_Vector_Add/source.cpp

```diff
@@ -6,6 +6,36 @@
 
  You should have received a copy of the license along with this
  work.  If not, see <http://creativecommons.org/licenses/by-sa/4.0/>.
+
+ * SYCL Quick Reference
+ * ~~~~~~~~~~~~~~~~~~~~
+ *
+ * // Default construct a queue
+ * auto q = sycl::queue{};
+ *
+ * // Declare a buffer pointing to ptr
+ * auto buf = sycl::buffer{ptr, sycl::range{n}};
+ *
+ * // Submit work to the queue
+ * q.submit([&](sycl::handler &cgh) {
+ *   // COMMAND GROUP
+ * });
+ *
+ * // Within the command group you can
+ * //    1. Declare an accessor to a buffer
+ *          auto read_write_acc = sycl::accessor{buf, cgh};
+ *          auto read_acc = sycl::accessor{buf, cgh, sycl::read_only};
+ *          auto write_acc = sycl::accessor{buf, cgh, sycl::write_only};
+ *          auto no_init_acc = sycl::accessor{buf, cgh, sycl::no_init};
```

Comment:
It is not clear what is the value of using the `auto` notation besides making the line longer. Otherwise there is `get_access`... ;-)

Reply:
I am a little bit biased - personally I feel that this notation is the easiest to read and make sense of. If we write:
```
sycl::accessor read_acc {buf, cgh, sycl::read_only};
```
I think the name of the variable is not as readable.

Having an 
```
auto var_name = class{};
``` 
Notation is very consistent and readable. The start of the var name is always at the same (relative) col and always before an equals sign. I know this is a bit pedantic but I think it is justified! :)

Reply:
Also for learning reasons we only want to use auto if the type of the obj is self evident. So something like:
```
auto acc = buf.get_access<mode>();
```
Is not very useful. Moreover using the accessor constructor instead of `get_access` allows us to use CTAD, which is modern and clean!

Reply:
Yes you can write as you want anyway, at the end this is your code. :-)
Just that the `sycl::accessor` has been improved to have CTAD in SYCL 2020 often terser than the `auto` syntax with `get_access`.